HTTP(S) proxy won't work under windows platform

**Describe the bug**
There's no way to use proxy smoothly, even i tried:
```
vcpkg env
set HTTPS_PROXY=https://127.0.0.1:10809/
set HTTP_PROXY=https://127.0.0.1:10809/
vcpkg install boost:x64-windows
```
The vcpkg says:
```
Downloading https://raw.githubusercontent.com/boostorg/boost/boost-1.75.0.beta1/boostcpp.jam... Failed. Status: 4;"A requested feature, protocol or option was not found built-in in this libcurl due to a build-time decision."
```
Seems the curl won't use windows proxy.
Isn't there any way for you to use another download tool on Windows for simple proxy setting like 
`vcpkg --proxy=https://127.0.0.1:10809` ? 

**Environment**
- OS: Windows
- Compiler: Visual Studio 2019 MSVC

**To Reproduce**
Steps to reproduce the behavior:
1.Start an HTTP proxy at 10809
2.
```
vcpkg env
set HTTPS_PROXY=https://127.0.0.1:10809/
set HTTP_PROXY=https://127.0.0.1:10809/
vcpkg install boost:x64-windows
```


**Expected behavior**
The vcpkg uses my HTTPS proxy to download files fast and easy.

**Failure logs** 
```
Starting package 1/121: boost-build:x64-windows
Building package boost-build[core]:x64-windows...
-- Using cached E:/vcpkg/downloads/boostorg-build-boost-1.75.0.beta1.tar.gz
-- Cleaning sources at E:/vcpkg/buildtrees/boost-build/src/75.0.beta1-4371cdd5cf.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source E:/vcpkg/downloads/boostorg-build-boost-1.75.0.beta1.tar.gz
-- Applying patch fix_options.patch
-- Using source at E:/vcpkg/buildtrees/boost-build/src/75.0.beta1-4371cdd5cf.clean
-- Using cached E:/vcpkg/downloads/boost_LICENSE_1_0.txt
-- Downloading https://raw.githubusercontent.com/boostorg/boost/boost-1.75.0.beta1/boostcpp.jam...
-- Downloading https://raw.githubusercontent.com/boostorg/boost/boost-1.75.0.beta1/boostcpp.jam... Failed. Status: 4;"A requested feature, protocol or option was not found built-in in this libcurl due to a build-time decision."
CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:184 (message):

      Failed to download file.
      If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment
      variables to "https://user:password@your-proxy-ip-address:port/".
      Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues

Call Stack (most recent call first):
  ports/boost-build/portfile.cmake:25 (vcpkg_download_distfile)
  scripts/ports.cmake:136 (include)


Error: Building package boost-build:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: boost-build:x64-windows
  Vcpkg version: 2020.11.12-nohash

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
No

--- a/scripts/cmake/vcpkg_download_distfile.cmake
+++ b/scripts/cmake/vcpkg_download_distfile.cmake
@@ -186,6 +186,10 @@ function(vcpkg_download_distfile VAR)
                 "    Failed to download file.\n"
                 "    If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment\n"
                 "    variables to \"https://user:password@your-proxy-ip-address:port/\".\n"
+                "    \n"
+                "    If error with status 4 (Issue #15434),\n"
+                "    try setting \"http://user:password@your-proxy-ip-address:port/\".\n"
+                "    \n"
                 "    Otherwise, please submit an issue at https://github.com/Microsoft/vcpkg/issues\n")
             else()
                 test_hash("${download_file_path_part}" "downloaded file" "The file may have been corrupted in transit. This can be caused by proxies. If you use a proxy, please set the HTTPS_PROXY and HTTP_PROXY environment variables to \"https://user:password@your-proxy-ip-address:port/\".\n")

